mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -732,51 +736,73 @@ void dispatchShouldNotSendToGroupListenerWhenError() {
 
             rabbitMQExtension.getRabbitMQ().pause();
 
-            try {
-                eventBus().dispatch(EVENT, NO_KEYS).block();
-            } catch (Exception e) {
-                // ignore
-            }
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
 
-            getSpeedProfile().longWaitCondition()
-                .untilAsserted(() -> assertThat(eventCollector.getEvents()).isEmpty());
+            assertThat(eventCollector.getEvents()).isEmpty();
         }
 
         @Test
-        void dispatchShouldNotPersistEventWhenDispatchingNoKeyGetError() {
+        void dispatchShouldPersistEventWhenDispatchingNoKeyGetError() {
             EventCollector eventCollector = eventCollector();
             eventBus().register(eventCollector, GROUP_A);
 
             rabbitMQExtension.getRabbitMQ().pause();
 
-            try {
-                eventBus().dispatch(EVENT, NO_KEYS).block();
-            } catch (Exception e) {
-                // ignore
-            }
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
 
-            getSpeedProfile().longWaitCondition()
-                .untilAsserted(() ->
-                    assertThat(deadLetter().containEvents().block()).isFalse());
+            assertThat(dispatchingFailureEvents()).containsOnly(EVENT);
         }
 
         @Test
-        void dispatchShouldNotPersistEventWhenDispatchingWithKeysGetError() {
+        void dispatchShouldPersistEventWhenDispatchingWithKeysGetError() {
             EventCollector eventCollector = eventCollector();
             eventBus().register(eventCollector, GROUP_A);
             eventBus().register(eventCollector, KEY_1);
 
             rabbitMQExtension.getRabbitMQ().pause();
 
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
+
+            assertThat(dispatchingFailureEvents()).containsOnly(EVENT);
+        }
+
+        @Test
+        void dispatchShouldPersistOnlyOneEventWhenDispatchingMultiGroupsGetError() {
+            EventCollector eventCollector = eventCollector();
+            eventBus().register(eventCollector, GROUP_A);
+            eventBus().register(eventCollector, GROUP_B);
+
+            rabbitMQExtension.getRabbitMQ().pause();
+
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
+
+            assertThat(dispatchingFailureEvents()).containsOnly(EVENT);
+        }
+
+        @Test
+        void dispatchShouldPersistEventsWhenDispatchingGroupsGetErrorMultipleTimes() {
+            EventCollector eventCollector = eventCollector();
+            eventBus().register(eventCollector, GROUP_A);
+
+            rabbitMQExtension.getRabbitMQ().pause();
+            doQuietly(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
+            doQuietly(() -> eventBus().dispatch(EVENT_2, NO_KEYS).block());
```

Comment:
What if I dispatch 2 time the same event?

It should be in deadLetter twice...